components/backup-stream/src/event_loader.rs

```diff
@@ -326,32 +341,28 @@ where
     }
 
     /// initialize a range: it simply scan the regions with leader role and send them to [`initialize_region`].
-    pub fn initialize_range(
-        &self,
-        start_key: Vec<u8>,
-        end_key: Vec<u8>,
-        start_ts: TimeStamp,
-    ) -> Result<Statistics> {
+    pub fn initialize_range(&self, start_key: Vec<u8>, end_key: Vec<u8>) -> Result<()> {
         let mut pager = RegionPager::scan_from(self.regions.clone(), start_key, end_key);
-        let mut total_stat = StatisticsSummary::default();
         loop {
             let regions = pager.next_page(8)?;
             debug!("scanning for entries in region."; "regions" => ?regions);
             if regions.is_empty() {
                 break;
             }
             for r in regions {
-                let handle = ObserveHandle::new();
-                self.tracing
-                    .register_region(&r.region, handle.clone(), Some(start_ts));
-                let region_id = r.region.get_id();
-                let snap = self.observe_over_with_retry(&r.region, move || {
-                    ChangeObserver::from_cdc(region_id, handle.clone())
-                })?;
-                let stat = self.do_initial_scan(&r.region, start_ts, snap)?;
-                total_stat.add_statistics(&stat);
+                // Note: Even we did the initial scanning, and blocking resolved ts from advancing,
+                //       if the next_backup_ts was updated in some extreme condition, there is still little chance to lost data:
+                //       For example, if a region cannot elect the leader for long time. (say, net work partition)
+                //       At that time, we have nowhere to record the lock status of this region.
```

Comment:
Can we have a alert about this situation?

Reply:
Generally, it's a little hard to detect this condition internal the TiKV node, because we don't have the integral global region status internal the store. But the `Observe Region Count` metric can indicate this in some way, maybe we can config the alert manager then.